core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -11264,4 +11264,44 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
 
     assertThat(toSql(root, DatabaseProduct.SPARK.getDialect()), isLinux(expectedSparkSql));
   }
+
+  @Test public void testCoalesceFunctionWithIntegerAndStringArgument() {
+    final RelBuilder builder = relBuilder();
+
+    final RexNode formatIntegerCastRexNode =
+        builder.call(SqlLibraryOperators.FORMAT,
+            builder.literal("'%11d'"), builder.scan("EMP").field(0));
+    final RexNode formatIntegerRexNode =
```

Comment:
Please rename the variable, as this is not a formatIntegerRexNode. Its a rex node for COALESCE. Same for below.

Reply:
Done